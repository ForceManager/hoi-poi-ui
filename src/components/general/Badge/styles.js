export default (theme) => ({
    root: {
        ...theme.typography.smallText,
        lineHeight: '20px',
        color: theme.colors.text.black,
        borderRadius: '3px',
        display: 'inline-block',
        padding: '0 10px',
        minWidth: 70,
        textAlign: 'center',
    },
    Text: {
        lineHeight: '20px',
        fontWeight: 500,
        textTransform: 'uppercase',
    },
    // Types
    error: {
        backgroundColor: theme.colors.redLight,
        '& $Text': {
            color: theme.colors.text.red,
        },
    },
    info: {
        backgroundColor: theme.colors.greyLight,
        '& $Text': {
            color: theme.colors.text.greySoft,
        },
    },
    success: {
        backgroundColor: theme.colors.greenLight,
        '& $Text': {
            color: theme.colors.text.green,
        },
    },
    warn: {
        backgroundColor: theme.colors.yellowLight,
        '& $Text': {
            color: theme.colors.text.yellow,
        },
    },
});
