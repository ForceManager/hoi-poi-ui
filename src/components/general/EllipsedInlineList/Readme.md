Default:

```jsx
const items = [
    'Amelia Matthews',
    'Leo Rodriguez',
    'Maya Patel',
    'Oscar Kim',
    'Alice Wong',
    'Lucas Foster',
    'Ava Thompson',
    'Gabriel Gonzalez',
    'Hazel Carter',
    'Isaac Davis',
];

<EllipsedInlineList items={items} />;
```

With custom postComponent:

```jsx
import { Tooltip, Text } from 'hoi-poi-ui';

const items = [
    'Amelia Matthews',
    'Leo Rodriguez',
    'Maya Patel',
    'Oscar Kim',
    'Alice Wong',
    'Lucas Foster',
    'Ava Thompson',
    'Gabriel Gonzalez',
    'Hazel Carter',
    'Isaac Davis',
];

const PostComponent = ({ count, className: className }) => {
    return (
        <Tooltip content={<span>Hello!</span>}>
            <Text color="blue500" className={className}>
                +{count}
            </Text>
        </Tooltip>
    );
};

<EllipsedInlineList items={items} postComponent={PostComponent} />;
```
