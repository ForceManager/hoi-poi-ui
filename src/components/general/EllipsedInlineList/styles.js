export default (theme) => ({
    root: {
        display: 'flex',
        flexFlow: 'row nowrap',
        gap: 4,
    },
    list: {
        display: 'block',
        flex: 1,
        '& > span': {
            display: 'inline',
            whiteSpace: 'pre',
        },
    },
    count: {
        flexShrink: 0,
        cursor: 'pointer',
    },
});