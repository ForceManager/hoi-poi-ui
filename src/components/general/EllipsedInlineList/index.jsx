import React, { memo, useCallback, useEffect, useMemo, useState, useRef } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { getOverrides, useClasses } from '../../../utils/overrides';
import { createUseStyles } from '../../../utils/styles';
import Text from '../../typography/Text';
import PostComponent from './components/PostComponent';

import styles from './styles';

const useStyles = createUseStyles(styles, 'EllipsedInlineList');

const EllipsedInlineList = ({
    className: classNameProp,
    classes: classesProp,
    items: list,
    overrides: overridesProp,
    postComponent: customPostComponent,
    separator = ',\u0020',
}) => {
    const [isTruncated, setIsTruncated] = useState(false);
    const [overflowCount, setOverflowCount] = useState(0);

    const containerRef = useRef();
    const resizeObserver = useRef(null);

    const classes = useClasses(useStyles, classesProp);
    const override = getOverrides(overridesProp, EllipsedInlineList.overrides);

    const handleEllipsis = useCallback(() => {
        if (!containerRef.current) return;
        
        const container = containerRef.current;
        const containerWidth = container.offsetWidth;
        const items = container.children;
        let itemsWidth = 0;
        let overflowItemsCount = 0;

        for (let i = 0; i < items.length; i++) {
            itemsWidth += items[i].offsetWidth;
            if (itemsWidth > containerWidth) {
                overflowItemsCount = items.length - i;
                break;
            }
        }

        setOverflowCount(overflowItemsCount);
        setIsTruncated(overflowItemsCount > 0);
    }, []);

    useEffect(() => {
        handleEllipsis();

        resizeObserver.current = new ResizeObserver(handleEllipsis);
        resizeObserver.current.observe(containerRef.current);

        return () => {
            if (resizeObserver.current) {
                resizeObserver.current.disconnect();
            }
        };
    }, [handleEllipsis]);

    const containerProps = useMemo(
        () => ({
            className: classNames(classes.root, classNameProp),
            ...override.root,
        }),
        [classNameProp, classes.root, override.root],
    );

    const listProps = useMemo(
        () => ({
            ref: containerRef,
            className: classes.list,
            isTruncated: true,
            ...override.List,
        }),
        [classes.list, override.List],
    );

    const countProps = useMemo(
        () => ({
            className: classes.count,
            color: 'orange500',
            count: overflowCount,
            ...override.Count,
        }),
        [classes.count, override.Count, overflowCount],
    );

    const FinalPostComponent = customPostComponent || PostComponent;

    return (
        <div {...containerProps}>
            <Text {...listProps}>
                {list.map((item, index) => (
                    <span key={index}>
                        {item}
                        {index === list.length - 1 ? '' : separator}
                    </span>
                ))}
            </Text>
            {isTruncated && <FinalPostComponent {...countProps} />}
        </div>
    );
};

EllipsedInlineList.overrides = ['root', 'List', 'Count'];

EllipsedInlineList.defaultProps = {
    separator: ',\u0020',
};

EllipsedInlineList.propTypes = {
    className: PropTypes.string,
    separator: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.string).isRequired,
    overrides: PropTypes.object,
    postComponent: PropTypes.elementType,
};

export default memo(EllipsedInlineList);
