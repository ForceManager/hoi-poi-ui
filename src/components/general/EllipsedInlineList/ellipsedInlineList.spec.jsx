import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { HoiPoiProvider } from '../../../utils/styles';
import EllipsedInlineList from '../EllipsedInlineList';
import Text from '../../typography/Text';
import Tooltip from '../../utils/Tooltip';

// Define a dummy implementation of the ResizeObserver class
class ResizeObserverMock {
    observe() {}
    unobserve() {}
    disconnect() {}
}

describe('EllipsedInlineList', () => {

    window.ResizeObserver = ResizeObserverMock;

    test('default match', () => {
        const items = [
            'Amelia Matthews',
            'Leo Rodriguez',
            'Maya Patel',
            'Oscar Kim',
            'Alice Wong',
            'Lucas Foster',
            'Ava Thompson',
            'Gabriel Gonzalez',
            'Hazel Carter',
            'Isaac Davis',
        ];


        const wrapper = mount(
            <HoiPoiProvider>
                <EllipsedInlineList items={items} />
            </HoiPoiProvider>,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('with custom postcomponent match', () => {
        const items = [
            'Amelia Matthews',
            'Leo Rodriguez',
            'Maya Patel',
            'Oscar Kim',
            'Alice Wong',
            'Lucas Foster',
            'Ava Thompson',
            'Gabriel Gonzalez',
            'Hazel Carter',
            'Isaac Davis',
        ];

        const PostComponent = ({ count, className }) => {
            return (
                <Tooltip content={<span>Hello!</span>}>
                    <Text color="blue500" className={className}>
                        +{count}
                    </Text>
                </Tooltip>
            );
        };

        const wrapper = mount(
            <HoiPoiProvider>
                <EllipsedInlineList items={items} postComponent={PostComponent} />
            </HoiPoiProvider>,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
