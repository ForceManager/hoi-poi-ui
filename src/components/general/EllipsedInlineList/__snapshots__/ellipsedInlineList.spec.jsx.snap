// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EllipsedInlineList default match 1`] = `
<Component>
  <ThemeProvider
    theme={
      {
        "colors": {
          "aqua100": "#e4fafd",
          "aqua200": "#cbf3fc",
          "aqua300": "#AEE5F8",
          "aqua400": "#98d5f1",
          "aqua500": "#76bde8",
          "aqua600": "#5695c7",
          "aqua700": "#3b71a7",
          "aqua900": "#16386f",
          "black": "#3c4248",
          "blue100": "#d3e3fe",
          "blue200": "#a7c5fe",
          "blue300": "#7BA4FE",
          "blue400": "#5a89fd",
          "blue500": "#245dfd",
          "blue600": "#1a47d9",
          "blue700": "#1234b6",
          "blue900": "#061879",
          "green100": "#E9FAD2",
          "green200": "#CFF6A6",
          "green300": "#A7E675",
          "green400": "#7FCD50",
          "green500": "#4CAD20",
          "green600": "#369417",
          "green700": "#237c10",
          "green900": "#095306",
          "neutral100": "#f9f9fa",
          "neutral150": "#f6f7f9",
          "neutral200": "#f4f5f6",
          "neutral300": "#e9ecee",
          "neutral400": "#dfe2e5",
          "neutral500": "#d4d8dc",
          "neutral600": "#a9b1b9",
          "neutral700": "#7d8a96",
          "neutral800": "#526373",
          "neutral900": "#273c50",
          "neutralBase": "#ffffff",
          "orange100": "#fff1cc",
          "orange200": "#ffde99",
          "orange300": "#FFC766",
          "orange400": "#ffb13f",
          "orange500": "#ff8c00",
          "orange600": "#db6e00",
          "orange700": "#b75400",
          "orange900": "#7a2c00",
          "overlay": "rgba(33,33,33,0.55)",
          "purple100": "#fae3ed",
          "purple200": "#f6c8e1",
          "purple300": "#E5A5CE",
          "purple400": "#cc84b8",
          "purple500": "#aa5b9c",
          "purple600": "#92428b",
          "purple700": "#792d7a",
          "purple900": "#441151",
          "red100": "#ffe5d5",
          "red200": "#ffc5ac",
          "red300": "#FF9E82",
          "red400": "#ff7863",
          "red500": "#ff3b30",
          "red600": "#db2328",
          "red700": "#b7182a",
          "red900": "#7a0929",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "turquoise100": "#c9fbf1",
          "turquoise200": "#95f8ec",
          "turquoise300": "#5EEBE5",
          "turquoise400": "#36d2d8",
          "turquoise500": "#00a8be",
          "turquoise600": "#0083a3",
          "turquoise700": "#006388",
          "turquoise900": "#00335b",
          "yellow100": "#fdf6cb",
          "yellow200": "#fcec97",
          "yellow300": "#F8DC63",
          "yellow400": "#f1cb3c",
          "yellow500": "#e9b200",
          "yellow600": "#c89400",
          "yellow700": "#a77700",
          "yellow900": "#6f4a00",
        },
        "effects": {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "boxShadow24": {
            "boxShadow": "0px 18px 40px rgba(39, 60, 80, 0.4)",
          },
          "buttonActive": {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": {
          "badges": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 10,
            "fontWeight": 400,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "bigText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "body": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 400,
            "letterSpacing": 0.6,
            "lineHeight": "20px",
          },
          "body1": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": 400,
            "letterSpacing": 0.5,
            "lineHeight": "28px",
          },
          "button": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0.6,
            "lineHeight": "16px",
          },
          "caption": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 400,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "captionMedium": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 500,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "defaultText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "letterSpacing": 0.5,
          },
          "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
          "fontSizeBase": 14,
          "fontSizeSmall": 11,
          "h1": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 96,
            "fontWeight": 300,
            "letterSpacing": -1.5,
            "lineHeight": "112px",
          },
          "h2": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 60,
            "fontWeight": 300,
            "letterSpacing": -0.5,
            "lineHeight": "71px",
          },
          "h3": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 48,
            "fontWeight": 400,
            "lineHeight": "57px",
          },
          "h4": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 34,
            "fontWeight": 400,
            "letterSpacing": 0.25,
            "lineHeight": "40px",
          },
          "h5": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 24,
            "fontWeight": 400,
            "lineHeight": "28px",
          },
          "h6": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "fontWeight": 500,
            "letterSpacing": 0.25,
            "lineHeight": "24px",
          },
          "largeText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "overline": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 500,
            "letterSpacing": 2,
            "lineHeight": "16px",
            "textTransform": "uppercase",
          },
          "smallText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
          "subtitle": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0.09,
            "lineHeight": "20px",
          },
          "subtitle1": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": 400,
            "letterSpacing": 0.15,
            "lineHeight": "24px",
          },
        },
        "utils": {
          "scrollbar": {
            "&::-webkit-scrollbar": {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": {
              "background": "#dfe2e5",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": {
              "background": "#dfe2e5",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
          "selection": {
            "& ::-moz-selection": {
              "backgroundColor": "#ffde99",
            },
            "& ::selection": {
              "backgroundColor": "#ffde99",
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": {
              "minify": false,
            },
            "plugins": [],
          },
          "plugins": PluginsRegistry {
            "plugins": {
              "external": [
                {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                {
                  "onProcessRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": [
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": {
              "onChangeValue": [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": [],
              "onProcessStyle": [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": [
                [Function],
              ],
            },
          },
          "version": "10.10.0",
        }
      }
    >
      <EllipsedInlineList
        items={
          [
            "Amelia Matthews",
            "Leo Rodriguez",
            "Maya Patel",
            "Oscar Kim",
            "Alice Wong",
            "Lucas Foster",
            "Ava Thompson",
            "Gabriel Gonzalez",
            "Hazel Carter",
            "Isaac Davis",
          ]
        }
        separator=", "
      >
        <div
          className="HoiPoi__EllipsedInlineList__root-1-0"
        >
          <ForwardRef
            className="HoiPoi__EllipsedInlineList__list-1-1"
            isTruncated={true}
            overrides={{}}
            type="body"
          >
            <span
              className="HoiPoi__Text__root-1-3 HoiPoi__EllipsedInlineList__list-1-1 HoiPoi__Text__body-1-13 HoiPoi__Text__truncated-1-19"
              style={{}}
            >
              <span
                key="0"
              >
                Amelia Matthews
                , 
              </span>
              <span
                key="1"
              >
                Leo Rodriguez
                , 
              </span>
              <span
                key="2"
              >
                Maya Patel
                , 
              </span>
              <span
                key="3"
              >
                Oscar Kim
                , 
              </span>
              <span
                key="4"
              >
                Alice Wong
                , 
              </span>
              <span
                key="5"
              >
                Lucas Foster
                , 
              </span>
              <span
                key="6"
              >
                Ava Thompson
                , 
              </span>
              <span
                key="7"
              >
                Gabriel Gonzalez
                , 
              </span>
              <span
                key="8"
              >
                Hazel Carter
                , 
              </span>
              <span
                key="9"
              >
                Isaac Davis
              </span>
            </span>
          </ForwardRef>
        </div>
      </EllipsedInlineList>
    </JssProvider>
  </ThemeProvider>
</Component>
`;

exports[`EllipsedInlineList with custom postcomponent match 1`] = `
<Component>
  <ThemeProvider
    theme={
      {
        "colors": {
          "aqua100": "#e4fafd",
          "aqua200": "#cbf3fc",
          "aqua300": "#AEE5F8",
          "aqua400": "#98d5f1",
          "aqua500": "#76bde8",
          "aqua600": "#5695c7",
          "aqua700": "#3b71a7",
          "aqua900": "#16386f",
          "black": "#3c4248",
          "blue100": "#d3e3fe",
          "blue200": "#a7c5fe",
          "blue300": "#7BA4FE",
          "blue400": "#5a89fd",
          "blue500": "#245dfd",
          "blue600": "#1a47d9",
          "blue700": "#1234b6",
          "blue900": "#061879",
          "green100": "#E9FAD2",
          "green200": "#CFF6A6",
          "green300": "#A7E675",
          "green400": "#7FCD50",
          "green500": "#4CAD20",
          "green600": "#369417",
          "green700": "#237c10",
          "green900": "#095306",
          "neutral100": "#f9f9fa",
          "neutral150": "#f6f7f9",
          "neutral200": "#f4f5f6",
          "neutral300": "#e9ecee",
          "neutral400": "#dfe2e5",
          "neutral500": "#d4d8dc",
          "neutral600": "#a9b1b9",
          "neutral700": "#7d8a96",
          "neutral800": "#526373",
          "neutral900": "#273c50",
          "neutralBase": "#ffffff",
          "orange100": "#fff1cc",
          "orange200": "#ffde99",
          "orange300": "#FFC766",
          "orange400": "#ffb13f",
          "orange500": "#ff8c00",
          "orange600": "#db6e00",
          "orange700": "#b75400",
          "orange900": "#7a2c00",
          "overlay": "rgba(33,33,33,0.55)",
          "purple100": "#fae3ed",
          "purple200": "#f6c8e1",
          "purple300": "#E5A5CE",
          "purple400": "#cc84b8",
          "purple500": "#aa5b9c",
          "purple600": "#92428b",
          "purple700": "#792d7a",
          "purple900": "#441151",
          "red100": "#ffe5d5",
          "red200": "#ffc5ac",
          "red300": "#FF9E82",
          "red400": "#ff7863",
          "red500": "#ff3b30",
          "red600": "#db2328",
          "red700": "#b7182a",
          "red900": "#7a0929",
          "secondaryOverlay": "rgba(2,168,189,0.10)",
          "shadow": "rgba(0, 0, 0, 0.25)",
          "transparentBlack": "rgba(38, 43, 52, 0.8)",
          "turquoise100": "#c9fbf1",
          "turquoise200": "#95f8ec",
          "turquoise300": "#5EEBE5",
          "turquoise400": "#36d2d8",
          "turquoise500": "#00a8be",
          "turquoise600": "#0083a3",
          "turquoise700": "#006388",
          "turquoise900": "#00335b",
          "yellow100": "#fdf6cb",
          "yellow200": "#fcec97",
          "yellow300": "#F8DC63",
          "yellow400": "#f1cb3c",
          "yellow500": "#e9b200",
          "yellow600": "#c89400",
          "yellow700": "#a77700",
          "yellow900": "#6f4a00",
        },
        "effects": {
          "boxShadow": "0 3px 9px 0 rgba(0, 0, 0, 0.25)",
          "boxShadow24": {
            "boxShadow": "0px 18px 40px rgba(39, 60, 80, 0.4)",
          },
          "buttonActive": {
            "boxShadow": "inset 0 -2px 4px 0 rgba(51, 51, 51, 0.23)",
          },
        },
        "typography": {
          "badges": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 10,
            "fontWeight": 400,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "bigText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "letterSpacing": 0.5,
          },
          "body": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 400,
            "letterSpacing": 0.6,
            "lineHeight": "20px",
          },
          "body1": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": 400,
            "letterSpacing": 0.5,
            "lineHeight": "28px",
          },
          "button": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0.6,
            "lineHeight": "16px",
          },
          "caption": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 400,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "captionMedium": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 500,
            "letterSpacing": 0.4,
            "lineHeight": "16px",
          },
          "defaultText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "letterSpacing": 0.5,
          },
          "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
          "fontSizeBase": 14,
          "fontSizeSmall": 11,
          "h1": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 96,
            "fontWeight": 300,
            "letterSpacing": -1.5,
            "lineHeight": "112px",
          },
          "h2": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 60,
            "fontWeight": 300,
            "letterSpacing": -0.5,
            "lineHeight": "71px",
          },
          "h3": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 48,
            "fontWeight": 400,
            "lineHeight": "57px",
          },
          "h4": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 34,
            "fontWeight": 400,
            "letterSpacing": 0.25,
            "lineHeight": "40px",
          },
          "h5": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 24,
            "fontWeight": 400,
            "lineHeight": "28px",
          },
          "h6": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 20,
            "fontWeight": 500,
            "letterSpacing": 0.25,
            "lineHeight": "24px",
          },
          "largeText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "letterSpacing": 0.5,
          },
          "letterSpacing": 0.5,
          "lineHeightBase": 1.5,
          "overline": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 12,
            "fontWeight": 500,
            "letterSpacing": 2,
            "lineHeight": "16px",
            "textTransform": "uppercase",
          },
          "smallText": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 11,
            "letterSpacing": 0.5,
          },
          "subtitle": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0.09,
            "lineHeight": "20px",
          },
          "subtitle1": {
            "fontFamily": ""Roboto", "Helvetica Neue", Helvetica, Arial, sans-serif",
            "fontSize": 16,
            "fontWeight": 400,
            "letterSpacing": 0.15,
            "lineHeight": "24px",
          },
        },
        "utils": {
          "scrollbar": {
            "&::-webkit-scrollbar": {
              "height": 17,
              "width": 17,
            },
            "&::-webkit-scrollbar-thumb": {
              "background": "#dfe2e5",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
            "&::-webkit-scrollbar-thumb:window-inactive": {
              "background": "#dfe2e5",
              "backgroundClip": "content-box",
              "border": "4px solid rgba(0, 0, 0, 0)",
              "borderRadius": 10,
            },
          },
          "selection": {
            "& ::-moz-selection": {
              "backgroundColor": "#ffde99",
            },
            "& ::selection": {
              "backgroundColor": "#ffde99",
            },
          },
        },
      }
    }
  >
    <JssProvider
      classNamePrefix="HoiPoi__"
      generateId={[Function]}
      jss={
        Jss {
          "generateId": [Function],
          "id": 3,
          "options": {
            "Renderer": [Function],
            "createGenerateId": [Function],
            "id": {
              "minify": false,
            },
            "plugins": [],
          },
          "plugins": PluginsRegistry {
            "plugins": {
              "external": [
                {
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                  "onUpdate": [Function],
                },
                {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                {
                  "onProcessRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                  "onProcessRule": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onProcessRule": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onProcessStyle": [Function],
                },
              ],
              "internal": [
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onChangeValue": [Function],
                  "onCreateRule": [Function],
                  "onProcessStyle": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
                {
                  "onCreateRule": [Function],
                },
              ],
            },
            "registry": {
              "onChangeValue": [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onCreateRule": [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessRule": [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onProcessSheet": [],
              "onProcessStyle": [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "onUpdate": [
                [Function],
              ],
            },
          },
          "version": "10.10.0",
        }
      }
    >
      <EllipsedInlineList
        items={
          [
            "Amelia Matthews",
            "Leo Rodriguez",
            "Maya Patel",
            "Oscar Kim",
            "Alice Wong",
            "Lucas Foster",
            "Ava Thompson",
            "Gabriel Gonzalez",
            "Hazel Carter",
            "Isaac Davis",
          ]
        }
        postComponent={[Function]}
        separator=", "
      >
        <div
          className="HoiPoi__EllipsedInlineList__root-2-0"
        >
          <ForwardRef
            className="HoiPoi__EllipsedInlineList__list-2-1"
            isTruncated={true}
            overrides={{}}
            type="body"
          >
            <span
              className="HoiPoi__Text__root-2-3 HoiPoi__EllipsedInlineList__list-2-1 HoiPoi__Text__body-2-13 HoiPoi__Text__truncated-2-19"
              style={{}}
            >
              <span
                key="0"
              >
                Amelia Matthews
                , 
              </span>
              <span
                key="1"
              >
                Leo Rodriguez
                , 
              </span>
              <span
                key="2"
              >
                Maya Patel
                , 
              </span>
              <span
                key="3"
              >
                Oscar Kim
                , 
              </span>
              <span
                key="4"
              >
                Alice Wong
                , 
              </span>
              <span
                key="5"
              >
                Lucas Foster
                , 
              </span>
              <span
                key="6"
              >
                Ava Thompson
                , 
              </span>
              <span
                key="7"
              >
                Gabriel Gonzalez
                , 
              </span>
              <span
                key="8"
              >
                Hazel Carter
                , 
              </span>
              <span
                key="9"
              >
                Isaac Davis
              </span>
            </span>
          </ForwardRef>
        </div>
      </EllipsedInlineList>
    </JssProvider>
  </ThemeProvider>
</Component>
`;
